org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/http/retry/CustomHttpRequestRetryStrategy.java

```diff
@@ -136,6 +143,24 @@ public boolean retryRequest(
     return handleAsIdempotent(request) || detectStaleSocketChannel(exception);
   }
 
+  @Override
+  public boolean retryRequest(HttpResponse response, int execCount, HttpContext context) {
+    boolean skipRetryByHeaderSetOnResponse = isFinalResponse(response);
+    if (skipRetryByHeaderSetOnResponse) {
+      LOG.trace("Respect response header {} to skip retry", FINAL_RESPONSE_HEADER);
+      return false;
+    }
+
+    return super.retryRequest(response, execCount, context);
+  }
+
+  /**
+   * @return true if {@link #FINAL_RESPONSE_HEADER} is set for response (and thus retry should be skipped regardless of status code)
+   */
+  protected boolean isFinalResponse(HttpResponse response) {
+    return response.getLastHeader(FINAL_RESPONSE_HEADER) != null;
```

Comment:
I think it's ok to just check if the header exists, so `FINAL_RESPONSE_HEADER=false` would also result in `true`.
In this case you can also just write:
```suggestion
    return response.containsHeader(FINAL_RESPONSE_HEADER);
```
Maybe you can add a comment about this, that we ignore the value of the header and just check for its existence.

Reply:
Changed: https://github.com/eclipse-scout/scout.rt/pull/1689#event-19459976491